lib/shared/utils.js

```diff
@@ -81,3 +83,53 @@ exports.makeFlatStartCaseObject = (object) => {
   });
 
 };
+
+/**
+ * @summary Create an object clone with all absolute paths replaced with the path basename
+ * @function
+ * @public
+ *
+ * @param {Object} object - original object
+ * @returns {Object} transformed object
+ *
+ * @example
+ * const anonymized = utils.hideAbsolutePathsInObject({
+ *   path1: '/home/john/rpi.img',
+ *   simpleProperty: null,
+ *   nested: {
+ *     path2: '/home/john/another-image.img',
+ *     path3: 'yet-another-image.img',
+ *     otherProperty: false
+ *   }
+ * });
+ *
+ * console.log(anonymized);
+ * > {
+ * >   path1: 'rpi.img',
+ * >   simpleProperty: null,
+ * >   nested: {
+ * >     path2: 'another-image.img',
+ * >     path3: 'yet-another-image.img',
+ * >     otherProperty: false
+ * >   }
+ * > }
+ */
+exports.hideAbsolutePathsInObject = (object) => {
+  return _.deepMapValues(object, (value) => {
+    if (!_.isString(value)) {
+      return value;
+    }
+
+    // Don't alter disk devices, even though they appear as full paths
+    if (_.some([
+      _.startsWith(value, '/dev/'),
+      _.startsWith(value, '\\\\.\\')
+    ])) {
```

Comment:
Not really, but I don't have any strong opinions. Just to brainstorm, this could be also expressed as:

```js
_.some(_.map([
  '/dev/',
  '\\\\.\\'
], _.partial(_.startsWith, value))) {
    ...
}
```

Although its probably not so readable :P